src/python/pants/backend/python/typecheck/pyright/rules.py

```diff
@@ -194,31 +194,17 @@ async def pyright_typecheck_partition(
         ),
     )
 
-    source_roots_get = Get(
-        SourceRootsResult,
-        SourceRootsRequest,
-        SourceRootsRequest.for_files(root_sources.snapshot.files),
-    )
-
-    requirements_venv_pex, source_roots = await MultiGet(
-        requirements_venv_pex_get,
-        source_roots_get,
-    )
-
     # Patch the config file to use the venv directory from the requirements pex,
     # and add source roots to the `extraPaths` key in the config file.
-    unique_source_roots = FrozenOrderedSet(
-        [root.path for root in source_roots.path_to_root.values()]
-    )
     patched_config_digest = await _patch_config_file(
-        config_files, requirements_venv_pex.venv_rel_dir, unique_source_roots
+        config_files, requirements_venv_pex.venv_rel_dir, transitive_sources.source_roots
```

Comment:
`.source_roots` holds the transitive roots, but `SourceRootsRequest` for all the files doesn't? 

Yeahhh, wouldn't have found this bug myself. Thanks Benjy!

Reply:
The previous code computed the source roots from just the user-specified sources, not their transitive deps.